dune/xt/grid/intersection.hh

```diff
@@ -66,12 +67,18 @@ double diameter(const Intersection<G, I>& intersection)
 {
   auto max_dist = std::numeric_limits<typename G::ctype>::min();
   const auto& geometry = intersection.geometry();
-  for (auto i : Common::value_range(geometry.corners())) {
-    const auto xi = geometry.corner(i);
-    for (auto j : Common::value_range(i + 1, geometry.corners())) {
-      auto xj = geometry.corner(j);
-      xj -= xi;
-      max_dist = std::max(max_dist, xj.two_norm());
+  if (geometry.corners() == 1) { // this is the 1d case, return diameter of adjacent elements instead
```

Comment:
Why is the diameter of a 1d intersection (so basically a point?) the diameter of its adjacent elements?